src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -93,4 +93,13 @@ public static Appointment parseAppointmentDescription(String appointment) {
 
         return new Appointment(valueField, parseAppointmentDate(dateField));
     }
+
+    @Override
+    public int compareTo(Appointment appointment) {
+        if (appointment instanceof NullAppointment) {
+            return 0;
```

Comment:
```suggestion
            return -1;
```

A nitpick since it works just as well(?), but 0 implies that the Appointment and the compared NullAppointment are equal in the sorting order. compareTo() can return values of -1, 0 and 1.